### ISSUE TITLE
RouteRestriction’s “at” parameter doesn’t seem to work with relative path components

### ISSUE THREAD

**Issue by Cellane** [skelpo/JWTMiddleware#12]
In my projects, I try to make controllers agnostic to the entire path they’re serving, and be responsible ideally only for one path component.

This way, I have an `APIController` (creating the `/api` route group) that “mounts” a `V1Controller` (appending the `/v1` route group) that finally “mounts” the `UserController` (appending the `/user` route group, resulting in a final path of `/api/v1/user`).

`UserController` then defines a route for user registration (relative to the already created path `/api/v1/user`, like this:

```swift
admin.post(User.self, at: "register", use: registerHandler)
```

Normal stuff so far 😅Next, I’d want to protect this route so that only admins can create new user accounts (that’s the `admin` variable in the previous code snippet), and intuition would lead me to believe that this is the correct way of doing it:

```swift
let users = router.grouped("user")
let admin = users.grouped(
    RouteRestrictionMiddleware<UserStatus, Payload, User>(
        restrictions: [
            RouteRestriction(.POST, at: "register", allowed: [.admin])
        ],
        failureError: .unauthorized
    ),
    JWTVerificationMiddleware()
)
```

In reality, it seems that this doesn’t work the way I’d expect, and the correct restriction would be `RouteRestriction(.POST, at: "/api/v1/user/register", allowed: [.admin])`, requiring me to type the full path, thus undermining the effort to make controllers agnostic to the path they’re “mounted” to.

I wonder if it’s possible to make the `at` parameter “act” in a relative way to the router group it’s assigned to.

**Comment by calebkleveter** [comment]
I am looking at this issue. It's one of those things that should be simple, but it isn't due to various restrictions on Vapor's part. I'll keep working on getting this working, but I can't say how long it will be before it is released.

**Comment by Cellane** [comment]
I see – thank you very much! I saw the related (very short 😅) discussion on Vapor’s Discord channel, so maybe I should report this as an enhancement suggestion to Vapor instead first, and only after that this could be solved?

**Comment by calebkleveter** [comment]
Probably. Thanks for helping out on this!

**Comment by calebkleveter** [comment]
Solved in release [0.8.0](https://github.com/skelpo/JWTMiddleware/releases/tag/0.8.0)
